javaparser-core/src/main/java/com/github/javaparser/ast/expr/IntegerLiteralExpr.java

```diff
@@ -77,13 +77,13 @@ public boolean remove(Node node) {
 
     public int asInt() {
         String result = value.replaceAll("_", "");
-        if (result.startsWith("0x")) {
+        if (result.startsWith("0x") || result.startsWith("0X")) {
```

Comment:
Do you mean uppercase 'value' or try to uppercase the second char if the size is > 1 and its not a number :) ? Your sure if thats more readable than wo startsWith checks?